Composer/packages/extensions/visual-designer/src/components/nodes/utils.ts

```diff
@@ -31,3 +31,21 @@ export function getUserAnswersTitle($type: string): string {
 
   return '';
 }
+
+const truncateSDKType = $type => (typeof $type === 'string' ? $type.split('Microsoft.')[1] : '');
+
+/**
+ * Title priority: $designer.name > override title > title from sdk schema > $type suffix
+ * @param overrideTitle customized title
+ */
+export function generateSDKTitle(overrideTitle?: string) {
+  const titleFromUischema = overrideTitle;
+  return data => {
+    const $type = get(data, '$type');
+    const titleFrom$designer = get(data, '$designer.name');
+    const titleFromShared = get(ConceptLabels, [$type, 'title']);
+    const titleFrom$type = truncateSDKType($type);
+
+    return titleFrom$designer || titleFromUischema || titleFromShared || titleFrom$type;
+  };
+}
```

Comment:
This could live in the shared package I think.

Reply:
what about adding an `getSDKTitle` helper in 'shared'?
```ts
function getSDKTitle(data) {
    return designerTitle || schemaTitile || typeTitle;
}
```

Reply:
moved to shared lib